model/change_point_detection.go

```diff
@@ -389,14 +388,44 @@ func GetTotalPagesForChangePointsGroupedByVersion(ctx context.Context, env cedar
 		}
 		return int(math.Ceil(float64(res.Count) / float64(pageSize))), nil
 	}
-	return 0, errors.New("Not able to get count of total changepoints matching query")
+	return 0, nil
 }
 
-func appendAfterBaseGetChangePointsByVersionAgg(projectId string, additionalSteps ...bson.M) []bson.M {
+func appendAfterBaseGetChangePointsByVersionAgg(projectId, variantRegex, versionRegex, taskRegex, testRegex, measurementRegex string, threadLevels []int, additionalSteps ...bson.M) []bson.M {
+	matchStage := bson.M{
+		bsonutil.GetDottedKeyName(perfInfoKey, perfResultInfoProjectKey):  projectId,
+		bsonutil.GetDottedKeyName(perfInfoKey, perfResultInfoVariantKey):  bson.M{"$regex": variantRegex},
+		bsonutil.GetDottedKeyName(perfInfoKey, perfResultInfoVersionKey):  bson.M{"$regex": versionRegex},
+		bsonutil.GetDottedKeyName(perfInfoKey, perfResultInfoTaskNameKey): bson.M{"$regex": taskRegex},
+		bsonutil.GetDottedKeyName(perfInfoKey, perfResultInfoTestNameKey): bson.M{"$regex": testRegex},
+	}
+
+	if threadLevels != nil {
+		matchStage[bsonutil.GetDottedKeyName(perfInfoKey, perfResultInfoArgumentsKey, "thread_level")] = bson.M{
+			"$in": threadLevels,
+		}
+	}
```

Comment:
Looks like threadlevels will need to be a csv list of integers, rather than a regex, since they are persisted as ints.

Reply:
I meant for the rest API --> we wanted to pass a regex initially but since they are persisted as integers we instead should just pass a slice of ints to match agains (which is a csv list of ints when invoking the API).